demos/free_surface/implicit_top_bottom_free_surface.py

```diff
@@ -0,0 +1,66 @@
+from gadopt import *
+from implicit_free_surface import ImplicitFreeSurfaceModel
+from test_viscous_surface import run_benchmark
+
+
+class TopBottomImplicitFreeSurfaceModel(ImplicitFreeSurfaceModel):
+    # Test case from Section 3.1.2 of `An implicit free surface algorithm
+    # for geodynamical simulations', Kramer et al 2012.
+
+    name = "implicit-both"
+    bottom_free_surface = True
+
+    def __init__(self, dt_factor, nx=80, do_write=True, iterative_2d=False, cartesian=True):
+        self.dt_factor = dt_factor
+        self.rho_bottom = 2
+        self.zeta_error = 0
+        super().__init__(dt_factor, nx=nx, do_write=do_write, iterative_2d=iterative_2d, cartesian=cartesian)
+
+        if iterative_2d:
+            # Schur complement splitting leads to a nullspace in the velocity block.
+            # Adding a small absorption term bringing the vertical velocity to zero removes this nullspace
+            # and does not effect convergence provided that this term is small compared with the overall numerical error.
+            self.absorption_penalty()
```

Comment:
Maybe `self.absorption_penalty(dt_factor)`, since it only seems to be called during initialisation anyway?

Reply:
sounds good